src/operator/contrib/proposal.cu

```diff
@@ -456,9 +459,10 @@ class ProposalGPUOp : public Operator{
     float* workspace_proposals_ptr = NULL;
     FRCNN_CUDA_CHECK(cudaMalloc(&workspace_proposals_ptr, sizeof(float) * count * 5));
     Tensor<xpu, 2> workspace_proposals(workspace_proposals_ptr, Shape2(count, 5));
-    FRCNN_CUDA_CHECK(cudaMemcpy(workspace_proposals.dptr_,
-                                &anchors[0], sizeof(float) * anchors.size(),
-      cudaMemcpyHostToDevice));
+    cudaStream_t stream = mshadow::Stream<gpu>::GetStream(s);
+    FRCNN_CUDA_CHECK(cudaMemcpyAsync(workspace_proposals.dptr_,
```

Comment:
Here, memcpy is being scheduled to run on `stream`, but [line 471](https://github.com/apache/incubator-mxnet/blob/743c10c3d931b84de71a3731a69c6494921dd43d/src/operator/contrib/proposal.cu#L471) runs `ProposalGridKernel` on the default context. Unless I am missing something, this introduces a potential race condition: when `stream` does not match the default stream, `ProposalGridKernel` may run before `cudaMemcpyAsync` completes.

Reply:
No. The fact that `ProposalGridKernel` runs in default stream is a bug, I agree, but it is a performance bug, not a correctness one. The semantics of the default stream is that it synchronizes with all other streams, so there is not a race here.